homeassistant/components/fritzbox/sensor.py

```diff
@@ -29,9 +34,58 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
             entities.append(FritzBoxTempSensor(device, fritz))
             devices.add(device.ain)
 
+        if device.battery_level is not None:
+            entities.append(FritzBoxBatterySensor(device, fritz))
+            devices.add(f"{device.ain}_{DEVICE_CLASS_BATTERY}")
+
     async_add_entities(entities)
 
 
+class FritzBoxBatterySensor(SensorEntity):
+    """The entity class for Fritzbox battery sensors."""
+
+    def __init__(self, device, fritz):
+        """Initialize the sensor."""
+        self._device = device
+        self._fritz = fritz
+
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(FRITZBOX_DOMAIN, self._device.ain)},
+            "manufacturer": self._device.manufacturer,
+            "model": self._device.productname,
+            "sw_version": self._device.fw_version,
+        }
```

Comment:
Can you put this in a base class?

Reply:
I thought of that at first, too, but skipped this idea for this PR, because a central `FritzBoxEntity` class should be implemented and used also for the other platforms, but IMHO this would overload this PR.